pixeltable/exprs/inline_array.py

```diff
@@ -100,7 +100,7 @@ def _from_dict(cls, d: Dict, components: List[Expr]) -> Expr:
         assert 'elements' in d
         arg: List[Any] = []
         for idx, val in d['elements']:
-            if idx is not None:
+            if idx is not None and idx >= 0:  # Older schemas might have -1 instead of None
```

Comment:
You'd need to apply the same logic to InlineDict.

But let's talk about this. I understand it's a fix, but I don't love having stuff like that hang around the codebase forever.